apps/src/music/progress/MusicValidator.ts

```diff
@@ -152,6 +156,10 @@ export default class MusicValidator extends Validator {
       }
     });
 
+    this.checkConditionPlayedDifferentSoundsTogetherMultipleTimes(
```

Comment:
I wonder if it could make sense to eventually restructure `checkConditions` to just be a list of calls like this that each map to one of the available validation keys. Some would likely contain repeated logic, but with good comments it could serve as reasonable documentation. 

Relatedly, as we continue to add more criteria here for K5 and CSD3 will `checkConditions`  inevitably become more expensive? If these proposed inner functions could have awareness of the level configuration, they could also return early if we don't care about what they're checking.

Reply:
Great calls, and reinforce some feelings I got when writing this.  

I think splitting out the validations might make sense as a refactor going forward, and we could certainly have a library of helper functions to avoid duplicating too much.  It is verging on unwieldy to try and do a lot of separate validations in a single pass, and indeed this validation felt easier done in its own pass.  I also think we might be able to skip over the concept of "satisfied conditions" that then get matched to validations, because having to add separate "satisfied conditions" for intermediate numbers of sounds (in `addPlayedConditions`) also feels like it's getting a bit unwieldy.

I was also starting to get concerned about the potential expense of all these validations.  We can cap the number of events to help this a bit, and we can also be careful about the complexity of our algorithms, but you also make a great point that we could just check the validations we care about.

Indeed, it seems possible we could invert things and process only the validation(s) we care about, evaluating them lazily.  This could be a fun refactor.  Would love @sanchitmalhotra126's thoughts on this too.

Reply:
Agreed with all of the above. It would be great to revisit the validation system in general now that we have a bunch more validations and have been exploring using them in other labs (ex. Python lab with unit testing).